src/platform/unix/linux/mod.rs

```diff
@@ -26,18 +26,64 @@ use std::thread;
 
 const MAX_FDS_IN_CMSG: u32 = 64;
 
+const SCM_RIGHTS: c_int = 0x01;
+
+// The value Linux returns for SO_SNDBUF
+// is not the size we are actually allowed to use...
+// Empirically, we have to deduct 32 bytes from that.
+const RESERVED_SIZE: usize = 32;
+
+#[cfg(target_os="android")]
+const TEMP_FILE_TEMPLATE: &'static str = "/sdcard/servo/ipc-channel-shared-memory.XXXXXX";
+
+#[cfg(not(target_os="android"))]
+const TEMP_FILE_TEMPLATE: &'static str = "/tmp/ipc-channel-shared-memory.XXXXXX";
+
+
+#[cfg(target_os="linux")]
+type IovLen = usize;
+#[cfg(target_os="linux")]
+type SockLen = usize;
```

Comment:
Ah, so for some reason the POSIX committee decided to abuse `socklen_t` for something that is clearly *not* in any way related to socket address lengths; while `glibc` decided to stick with the more appropriate `size_t`... This is so retarded :-(

Calling the type `SockLen` is very confusing though, as it has no bearing on the primary use of `socklen_t` -- better call it `MsgControllen`, or something along these lines. Also, it's probably better to define it as `size_t` rather than `usize`, to match the `libc` definition.